src/ArrayFromString.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @since 0.5.17
+ */
+import * as t from 'io-ts'
+import { pipe } from 'fp-ts/lib/pipeable'
+import { map, Traversable } from 'fp-ts/lib/Array'
+import { chain, Applicative } from 'fp-ts/lib/Either'
+
+/**
+ * @since 0.5.17
+ */
+export interface ArrayFromStringC<C extends t.Mixed> extends t.Type<Array<t.TypeOf<C>>, string, unknown> {}
+
+/**
+ * @example
+ * import { ArrayFromString } from 'io-ts-types/lib/ArrayFromString'
+ * import { NumberFromString } from 'io-ts-types/lib/NumberFromString'
+ * import { right } from 'fp-ts/lib/Either'
+ *
+ * assert.deepStrictEqual(ArrayFromString(NumberFromString, ",").decode("1,2,3"), right([1,2,3]))
+ * assert.deepStrictEqual(ArrayFromString(NumberFromString, /[\s,]+/).decode("1, 2, 3"), right([1,2,3]))
+ * @since 0.5.17
+ */
+export function ArrayFromString<C extends t.Mixed>(
+  codec: C,
+  decoderSeparator: string | RegExp = '',
+  encoderSeparator = '',
```

Comment:
Defining different separators between decoder and encoder will break codec laws where `encode(decode(u)) = u` and `decode(encode(a)) = right(a)` should hold.

Reply:
Agree with @mlegenhausen that distinct separators for decoding and encoding are less than ideal. It's not "the simplest solution that might possibly work" and passing `undefined` below in the tests is neither type-safe nor easy to read.

I think these principles also permit the `string | RegExp` to be simplified to `string`, which will encourage more readable code with less opportunity for cleverness/abuse.

Reply:
@mantas-janulionis I would also go with @EricCrosson idea and change this to one `separator` of type `string` this limits the possibility for misusage.

> I think that this codec rule is little bit too strict. In my opinion, the library is supposed to deal with user input, which can be imperfect.

Actually it isn't supposed for that 😅 this is not a form validation library. This library is designed to make general input and output of your application type safe for example when receiving data from external systems and for this case you normally want a very strict validation. But yes user input can also be seen as external system but then I would combine it with a library that can normalize the data that comes from the user and for example can remove unnecessary spaces.